diff -upr a/midori/midori-view.c b/midori/midori-view.c
--- a/midori/midori-view.c
+++ b/midori/midori-view.c
@@ -38,10 +38,6 @@
     #include <sys/utsname.h>
 #endif
 
-/* This is unstable API, so we need to declare it */
-gchar*
-webkit_web_view_get_selected_text (WebKitWebView* web_view);
-
 static void
 midori_view_construct_web_view (MidoriView* view);
 
@@ -4166,10 +4162,26 @@ midori_view_get_link_uri (MidoriView* vi
 gboolean
 midori_view_has_selection (MidoriView* view)
 {
+    WebKitDOMDocument *doc;
+    WebKitDOMDOMWindow *window;
+    WebKitDOMDOMSelection *selection;
+    WebKitDOMRange *range;
+
     g_return_val_if_fail (MIDORI_IS_VIEW (view), FALSE);
 
-    katze_assign (view->selected_text, webkit_web_view_get_selected_text (
-        WEBKIT_WEB_VIEW (view->web_view)));
+    doc = webkit_web_view_get_dom_document (WEBKIT_WEB_VIEW (view->web_view));
+    window = webkit_dom_document_get_default_view (doc);
+
+    selection = webkit_dom_dom_window_get_selection (window);
+    if (selection == NULL)
+        return FALSE;
+
+    range = webkit_dom_dom_selection_get_range_at (selection, 0, NULL);
+    if (range == NULL)
+        return FALSE;
+
+    katze_assign (view->selected_text, webkit_dom_range_get_text (range));
+
     if (view->selected_text && *view->selected_text)
         return TRUE;
     else
